submissions/gurpreet-shawn/scripts/articleView.js

```diff
@@ -0,0 +1,129 @@
+    if (!$(this).hasClass('template')) {
+      // DONE: We need to take every author name from the page, and make it an option in the Author filter.
+      //       To do so, we build an `option` DOM element that we can append to the author select box.
+      //       Start by grabbing the author's name from `this` article element, and then use that bit of
+      //       text to create the option tag (in a variable named `optionTag`),
+      //       that we can append to the #author-filter select element.
+      //       YAY, DOM manipulation!
+      var val = $(this).find('address a').text();
+      var optionTag = '<option value="' + val + '">' + val + '</option>';
+      $('#author-filter').append(optionTag);
+
+      // DONE: Similar to the above, but...
+      //       Avoid duplicates! We don't want to append the category name if the select
+      //       already has this category as an option!
+      val = $(this).attr('data-category');
+      optionTag = '<option value="' + val + '">' + val + '</option>';
+      if ($('#category-filter option[value="' + val + '"]').length === 0) {
+        $('#category-filter').append(optionTag);
+      }
+    }
+  });
+};
+
+articleView.handleAuthorFilter = function() {
+  $('#author-filter').on('change', function() {
+    if ($(this).val()) {
+      // DONE: If the select box was changed to an option that has a value, we need to:
+      //       1. Hide all the articles,
+      //       2. Fade in only the articles that match based on the author that was selected.
+      //          Use an "attribute selector" to find those articles that match the value,
+      //          and fade them in for the reader.
+      var $value = $(this).val();
+      $('article').hide();
+      $('article').each(function(){
+        if ($(this).data('author')===$value){
+          $(this).show();
+        }
+      });
+    } else {
+      $('article').not('template').show();
+
+
+      // DONE: Otherwise, we should:
+      //       1. Show all the articles,
+      //       2. Except the one article we are using as a template.
+
+    }
+    // Reset the category-filter:
+    $('#category-filter').val('');
+  });
+};
+
+articleView.handleCategoryFilter = function() {
+  // TODO: Just like we do for #author-filter above, we should handle change
+  //       events on the #category-filter element. When an option with a value
+  //       is selected, hide all the articles, then reveal the matches.
+  //       When the blank (default) option is selected, show all the articles,
+  //       except for the template. Be sure to reset the #author-filter while you are at it!
+  $('#category-filter').on('change', function(){
+    if($(this).val()){
+      var $value = $(this).val();
+      $('article').hide();
+      $('article').each(function(){
+        if($(this).data('category') === $value){
+          $(this).show();
+        }
+      });
+    }
+    else{
+      $('article').not('template').show();
```

Comment:
template is showing when no author is selected.
this is close, template is a class, what symbol are classes referenced with?